src/App/Fossa/Ficus/Analyze.hs

```diff
@@ -123,20 +144,40 @@ ficusMessagesToFicusSnippetScanResults messages =
 runFicus ::
   ( Has Diagnostics sig m
   , Has (Lift IO) sig m
-  , Has ReadFS sig m
-  , Has Exec sig m
   , Has Logger sig m
   ) =>
   FicusConfig ->
   m FicusMessages
 runFicus ficusConfig = do
+  logDebugWithTime "About to extract Ficus binary..."
   withFicusBinary $ \bin -> do
+    logDebugWithTime "Ficus binary extracted, building command..."
     cmd <- ficusCommand ficusConfig bin
-    logDebug "Executing ficus"
-    result <- execThrow' cmd
-    let messages = parseFicusJson result Nothing
+    logDebugWithTime "Executing ficus (streaming)"
+    logDebug $ "Ficus command: " <> pretty (renderCommand cmd)
+    logDebug $ "Working directory: " <> pretty (toFilePath $ ficusConfigRootDir ficusConfig)
+
+    logDebugWithTime "Creating process configuration..."
+    let processConfig =
+          setWorkingDir (toFilePath $ ficusConfigRootDir ficusConfig) $
+            setStdout createPipe $
+              setStderr createPipe $
+                proc (toString $ cmdName cmd) (map toString $ cmdArgs cmd)
+
+    logInfo $ "Running Ficus analysis on " <> pretty (toFilePath $ ficusConfigRootDir ficusConfig)
+    logDebugWithTime "Starting Ficus process..."
+    messages <- sendIO $ withProcessWait processConfig $ \p -> do
```

Comment:
This seems OK, but I would still recommend using the [conduit form](https://hackage-content.haskell.org/package/conduit-extra-1.3.8/docs/Data-Conduit-Process.html#v:sourceProcessWithStreams) if you can, since it will do most of this book-keeping you're doing manually.

The idea is you can create a consumer like [mapM_](https://hackage-content.haskell.org/package/conduit-1.3.6.1/docs/Data-Conduit-Combinators.html#v:mapM_) for each of the stdout/stderr sinks. 

The only thing I'm not totally sure of is if our concrete `m`s implement `MonadUnliftIO`. If they don't, then the types may not match up. If that's an obstacle you ran into I can see what it might take to add those instances. Based on a quick read of the docs they should be derivable.